__tests__/ForgotPassword.js

```diff
@@ -0,0 +1,319 @@
+import '@testing-library/jest-dom/extend-expect';
+
+import { fireEvent, waitForDomChange } from '@testing-library/react';
+import { forceRemoveAll } from 'lib/flashMessage';
+import RoutePath from 'lib/routePath';
+import nock from 'nock';
+import { StatusCodes } from 'shared/constants';
+import { AppRoutes } from 'shared/constants/routes';
+import {
+  authTokenResponse,
+  AWSInfoResponse,
+  generateRandomString,
+  getMockCall,
+  postMockCall,
+  USER_EMAIL,
+  userResponse,
+} from 'testUtils/mockHttpCalls';
+import { renderRouteWithStore } from 'testUtils/renderUtils';
+
+describe('PasswordReset', () => {
+  beforeAll(() => {
+    nock.disableNetConnect();
+  });
+
+  afterAll(() => {
+    nock.enableNetConnect();
+  });
+
+  afterEach(() => {
+    if (!nock.isDone()) {
+      // eslint-disable-next-line no-console
+      console.error('Nock has pending mocks:', nock.pendingMocks());
+    }
+
+    expect(nock.isDone()).toBeTruthy();
```

Comment:
I'm trying these approaches in the tests I am writing/modifying and I think that `afterEach` callback should be an `async` function, and this expectation should be _awaited_:

```javascript
await wait(() => expect(nock.isDone()).toBe(true));
```

Otherwise you can experience different results or race conditions depending on the tests that runs before.

Reply:
Makes sense, I'll make the change 👍